### PR TITLE
tests: add integration tests

### PR DESCRIPTION
**Pull request by PierrickGT**

### CODE REVIEW

File: test/integration/Integration.t.sol [hunk 1]
```diff
@@ -0,0 +1,640 @@
...
+
+    function totalEarningSupply() external view returns (uint240 supply);
+
+    function totalNonEarningSupply() external view returns (uint240 supply);
+}
+
+contract IntegrationTests is Test {
+    IMTokenLike internal constant _mToken = IMTokenLike(0x866A2BF4E572CbcF37D5071A7a58503Bfb36be1b);
+
+    address internal constant _minterGateway = 0xf7f9638cb444D65e5A40bF5ff98ebE4ff319F04E;
```
Comment (PierrickGT) [inline test/integration/Integration.t.sol:38]: Unused.
```suggestion
```

File: test/integration/Integration.t.sol [hunk 2]
```diff
@@ -0,0 +1,640 @@
...
+        assertEq(_mToken.balanceOf(_dave), _daveBalance);
+        assertEq(_wrappedMToken.balanceOf(_dave), _daveBalance -= _daveBalance);
+        assertEq(_wrappedMToken.accruedYieldOf(_dave), 0);
+
+        // Assert Globals
+        assertEq(_wrappedMToken.totalEarningSupply(), 3);
+        assertEq(_wrappedMToken.totalNonEarningSupply(), 0);
+        assertEq(_wrappedMToken.totalSupply(), 3);
+        assertEq(_wrappedMToken.totalAccruedYield(), 4);
+        assertEq(_wrappedMToken.excess(), _excess -= 0);
```
Comment (PierrickGT) [inline test/integration/Integration.t.sol:589]:
```suggestion
        assertEq(_wrappedMToken.excess(), _excess);
```

File: test/integration/Integration.t.sol [hunk 3]
```diff
@@ -0,0 +1,640 @@
...
+
+        assertEq(_wrappedMToken.claimExcess(), _excess);
+        assertEq(_mToken.balanceOf(_vault), _excess + vaultStartingBalance_);
+
+        // Assert Globals
+        assertEq(_wrappedMToken.totalEarningSupply(), 3);
+        assertEq(_wrappedMToken.totalNonEarningSupply(), 0);
+        assertEq(_wrappedMToken.totalSupply(), 3);
+        assertEq(_wrappedMToken.totalAccruedYield(), 4);
+        assertEq(_wrappedMToken.excess(), _excess -= _excess);
```
Comment (PierrickGT) [inline test/integration/Integration.t.sol:603]:
```suggestion
        assertEq(_wrappedMToken.excess(), 0);
```

File: test/integration/Integration.t.sol (hunk 1 above)
Comment (deluca-mike) [reply test/integration/Integration.t.sol:38]: Wil be used when new integration tests are added that actually mint more M.

File: test/integration/Integration.t.sol (hunk 3 above)
Comment (deluca-mike) [reply test/integration/Integration.t.sol:603]: I think the original l;ine is better at conveying that all the excess tracked is being decremented by the axcess removed.

**Pull request merged**